### PR TITLE
Factor out modules shared among shim profiles [databricks]

### PR DESCRIPTION
**Pull request by gerashegalov**
- remove seemingly unused udf-compiler profile, this sub-module is used by all other profiles
- factor out shared sub-modules 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

**Review by revans2 — APPROVED** [review]
Does this make it so we cannot disable the udf-compiler any more? It looks like we couldn't really if we were doing any release profile. I am really just checking.

**Review by jlowe — APPROVED** [review]

**Comment by gerashegalov** [comment]
> Does this make it so we cannot disable the udf-compiler any more? It looks like we couldn't really if we were doing any release profile. I am really just checking.

Good question I don't know an answer to. Here is the original PR #434 where the module structure looked similar. @wjxiz1992  @abellina  can you chime in about the intent and the use of the `udf-compiler` profile?

File: pom.xml [hunk 1]
```diff
@@ -69,6 +69,16 @@
             <timezone>-6</timezone>
         </developer>
     </developers>
+    <!-- modules shared among profiles -->
+    <modules>
```
Comment (tgravescs) [inline pom.xml:73]: so we had common at one point then I think it was the spark 2.x support for explain that forced us to move within each profile.  That is gone so this should be good to put back.

**Review by tgravescs — APPROVED** [review]

**Comment by abellina** [comment]
> > Does this make it so we cannot disable the udf-compiler any more? It looks like we couldn't really if we were doing any release profile. I am really just checking.
> 
> Good question I don't know an answer to. Here is the original PR #434 where the module structure looked similar. @wjxiz1992 @abellina can you chime in about the intent and the use of the `udf-compiler` profile?

Since the udf compiler is a feature not recommended to just blindly turn on, I think we wanted some control to turn it on and off. It is included in our release bits and default disabled by configuration so I don't think this a big deal + this change doesn't preclude us from using `-pl "!udf-compiler"` to stop building it if we need to.

**Review by abellina — APPROVED** [review]

**Pull request merged**